src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -944,39 +941,58 @@ String getResponseToken(String responseInbox) {
         return responseInbox.substring(len);
     }
 
-    void cleanResponses(boolean cancelIfRunning) {
+    void cleanResponses(boolean closing) {

```

Comment:
change to this function takes advantage of NatsRequestCompletableFuture for better granularity and recognition of timeout